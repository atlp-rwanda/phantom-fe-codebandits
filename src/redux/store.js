import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer
  }
});
