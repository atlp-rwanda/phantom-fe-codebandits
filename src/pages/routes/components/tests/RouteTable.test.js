import React from 'react';
import renderer from 'react-test-renderer';
import {  mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import RouteTable from '../RouteTable.js'
import { Provider } from 'react-redux';
import {store} from '../../../../redux/store.js'


describe('Route table Page', () => {
  it('should render the Route table component', () => {
    const elem = renderer
      .create(
		  <Provider store={store}>
          <MemoryRouter>
            <RouteTable />
          </MemoryRouter>
		  </Provider>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render the button component', () => {
   const wrapper = mount(
	  <Provider store={store}>
          <MemoryRouter>
            <RouteTable />
          </MemoryRouter>
		  </Provider>
   )
   
   expect(wrapper.find('SearchFilter').length).toEqual(1)
  });
});
