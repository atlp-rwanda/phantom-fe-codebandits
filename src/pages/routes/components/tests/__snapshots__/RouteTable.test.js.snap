// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route table Page should render the Route table component 1`] = `
Array [
  <section
    className="flex justify-between content-center pb-7 font-raleway"
  >
    <div>
      <input
        className="p-1 rounded-md border-2"
        name="search"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
     
    
  </section>,
  " ",
  <table
    className="w-full"
    role="table"
  >
    <thead>
      <tr
        className="pb-10 "
        role="row"
      >
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Destination 1
        </th>
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Destination 2
        </th>
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Distance
        </th>
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Total buses in Route
        </th>
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Management
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      <tr
        role="row"
      >
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          Kabuga
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          Nyanza
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          22km
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          100
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          <div
            className="z-10"
          >
            <select
              className="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
              id=""
              name=""
              onChange={[Function]}
              role="select"
            >
              <option
                hidden={true}
              >
                 Manage 
              </option>
              <option
                value="view"
              >
                 View 
              </option>
              
              
               
            </select>
          </div>
        </td>
      </tr>
      <tr
        role="row"
      >
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          Kabuga
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          Nyabugogo
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          20km
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          100
        </td>
        <td
          className="pr-5 py-4 border-b-2"
          role="cell"
        >
          <div
            className="z-10"
          >
            <select
              className="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
              id=""
              name=""
              onChange={[Function]}
              role="select"
            >
              <option
                hidden={true}
              >
                 Manage 
              </option>
              <option
                value="view"
              >
                 View 
              </option>
              
              
               
            </select>
          </div>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
